anxcloud/resource_virtual_server.go

```diff
@@ -290,33 +289,17 @@ func resourceVirtualServerRead(ctx context.Context, d *schema.ResourceData, m in
 		return nil
 	}
 
-	err = resource.RetryContext(ctx, d.Timeout(schema.TimeoutCreate), func() *resource.RetryError {
-		info, infoErr := vsphereAPI.Info().Get(ctx, d.Id())
-		if infoErr != nil {
-			if err := handleNotFoundError(infoErr); err != nil {
-				return resource.NonRetryableError(fmt.Errorf("unable to get vm with id '%s': %w", d.Id(), err))
-			}
-			d.SetId("")
-			return nil
-		}
-		var nicErr error
-		for _, nic := range info.Network {
-			if len(nic.IPv4) == 0 && len(nic.IPv6) == 0 {
-				nicErr = multierror.Append(nicErr, fmt.Errorf("missing IPs for NIC"))
-			}
-		}
-		if nicErr != nil {
-			return resource.RetryableError(nicErr)
-		}
-		return nil
-	})
+	nicTypes, err := nicAPI.List(ctx)
```

Comment:
maybe we should move this a bit closer to where it is used, around line 346 if I see that correctly